token-metadata/program/src/processor/metadata/print.rs

```diff
@@ -0,0 +1,194 @@
+    let payer_info = ctx.accounts.payer_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let master_token_account_owner_info = ctx.accounts.master_token_account_owner_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let master_token_account_info = ctx.accounts.master_token_account_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let master_metadata_info = ctx.accounts.master_metadata_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let update_authority_info = ctx.accounts.update_authority_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let token_program = ctx.accounts.token_program_info;
+    let ata_program = ctx.accounts.ata_program_info;
+    let sysvar_instructions = ctx.accounts.sysvar_instructions_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let system_program = ctx.accounts.system_program_info;
+
+    // If the edition token account isn't already initialized, create it.
+    // If it does exist, validate it.
+    if edition_token_account_info.data_is_empty() {
+        // if the token account is empty, we will initialize a new one but it must
+        // be an ATA account
+        assert_derivation(
+            &spl_associated_token_account::id(),
+            edition_token_account_info,
+            &[
+                edition_token_account_owner_info.key.as_ref(),
+                spl_token::id().as_ref(),
+                edition_mint_info.key.as_ref(),
+            ],
+        )?;
+
+        // creating the associated token account
+        invoke(
+            &spl_associated_token_account::instruction::create_associated_token_account(
+                ctx.accounts.payer_info.key,
+                edition_token_account_owner_info.key,
+                edition_mint_info.key,
+                &spl_token::id(),
+            ),
+            &[
+                payer_info.clone(),
+                edition_token_account_owner_info.clone(),
+                edition_mint_info.clone(),
+                edition_token_account_info.clone(),
+            ],
+        )?;
+    } else {
+        assert_owned_by(edition_token_account_info, &spl_token::id())?;
+    }
+
+    if ata_program.key != &spl_associated_token_account::ID {
+        return Err(ProgramError::IncorrectProgramId);
+    }
+
+    if sysvar_instructions.key != &sysvar::instructions::ID {
+        return Err(ProgramError::IncorrectProgramId);
+    }
+
+    // Deserialize the master edition's metadata so we can determine token type
+    let master_metadata = Metadata::from_account_info(master_metadata_info)?;
+    let token_standard = master_metadata
+        .token_standard
+        .unwrap_or(TokenStandard::NonFungible);
```

Comment:
Is this assumption that accurate? Like, is it worth validating the master edition here to make sure?

Reply:
What is someone passed in an existing Print Edition that had a token standard of `None`, as the Master Edition?  I don't see a hardline check that we are in fact using a Master Edition.  Going down into the code, all I see is manual deserialization in functions such as `get_max_supply_off_master_edition`, which most likely fail if the account was an `Edition` rather than a `MasterEditionV2`, but still I haven't seen explicit check so it feels like using a side effect to guarantee safety.

As an alternative, in `Update` it is an explicit check using these utility functions: https://github.com/metaplex-foundation/metaplex-program-library/blob/master/token-metadata/program/src/utils/mod.rs#L44-L85

In conclusion, you might want to consider adding `is_master_edition` here which is what I also interpret from the previous comments as well.